crates/blockchain/metrics/rpc.rs

```diff
@@ -0,0 +1,85 @@
+use prometheus::{CounterVec, HistogramVec, register_counter_vec, register_histogram_vec};
+use std::{future::Future, sync::LazyLock};
+
+pub static METRICS_RPC_REQUEST_OUTCOMES: LazyLock<CounterVec> =
+    LazyLock::new(initialize_rpc_outcomes_counter);
+
+pub static METRICS_RPC_DURATION: LazyLock<HistogramVec> =
+    LazyLock::new(initialize_rpc_duration_histogram);
+
+// Metrics defined in this module register into the Prometheus default registry.
+// The metrics API exposes them by calling `gather_default_metrics()`.
+
+fn initialize_rpc_outcomes_counter() -> CounterVec {
+    register_counter_vec!(
+        "rpc_requests_total",
+        "Total number of RPC requests partitioned by namespace, method, and outcome",
+        &["namespace", "method", "outcome", "error_kind"],
+    )
+    .unwrap()
+}
+
+fn initialize_rpc_duration_histogram() -> HistogramVec {
+    register_histogram_vec!(
+        "rpc_request_duration_seconds",
+        "Histogram of RPC request handling duration partitioned by namespace and method",
+        &["namespace", "method"],
+    )
+    .unwrap()
+}
+
+/// Represents the outcome of an RPC request when recording metrics.
+#[derive(Clone)]
+pub enum RpcOutcome {
+    Success,
+    Error(String),
```

Comment:
This should enforce compile-time known strings: 
```suggestion
    Error(&'static str),
```